### PR TITLE
RTE comments implementation [1-4]+[6-9]

### PR DESCRIPTION
**Pull request by Milos-RTEi**

### CODE REVIEW

File: src/solver/optimisation/adequacy_patch.cpp [hunk 1]
```diff
@@ -254,13 +225,49 @@ std::pair<double, double> calculateAreaFlowBalance(PROBLEME_HEBDO* ProblemeHebdo
...
+                             ->ValeursHorairesDeDefaillanceNegative[hour];
+
+            netPositionInitValues[Area] = netPositionInit;
+            densNewValues[Area] = densNew;
+            rhsAreaBalanceValues[Area] = ensInit + netPositionInit - spillageInit;
+
+            logs.debug() << "DENS_new[" << Area << "] = " << densNewValues[Area];
+            logs.debug() << "rhsAreaBalanceValues[" << Area << "] = " << rhsAreaBalanceValues[Area]
+                         << " = ENSinit(" << ensInit << ") + NetPositionInit(" << netPositionInit
+                         << ") - SpillageInit(" << spillageInit << ")";
```
Comment (rtei-pw) [inline src/solver/optimisation/adequacy_patch.cpp]: We can remove the logs

File: src/solver/simulation/economy.cpp [hunk 2]
```diff
@@ -94,11 +94,20 @@ void AdequacyPatchOptimization::solve(Variable::State& state, int hourInTheYear,
     // part that we need
     ::SIM_RenseignementProblemeHebdo(*problemeHebdo, state, numSpace, hourInTheYear);
     OPT_OptimisationHebdomadaire(problemeHebdo, numSpace);
+
+    std::set<int> hoursRequiringCurtailmentSharing = getHoursRequiringCurtailmentSharing(numSpace);
+    for (int hourInWeek : hoursRequiringCurtailmentSharing)
+    {
+        logs.debug() << "========= [CSR]: Starting hourly optim for " << hourInWeek;
```
Comment (rtei-pw) [inline src/solver/simulation/economy.cpp]: we can remove the logs

File: src/solver/simulation/economy.cpp [hunk 3]
```diff
@@ -174,66 +183,42 @@ bool Economy::simulationBegin()
...
+    float threshold
+      = pProblemesHebdo[numSpace]->adqPatchParams->ThresholdInitiateCurtailmentSharingRule;
     std::set<int> triggerCsrSet;
     for (int i = 0; i < nbHoursInAWeek; ++i)
     {
-        if ((int)sumENS[i] >= threshold)
+        if ((int)sumENS[i] > threshold)
         {
+            logs.debug() << "hour: [" << i << "], sumENS = [" << (int)sumENS[i]
+                         << "], threshold = " << threshold;
```
Comment (rtei-pw) [inline src/solver/simulation/economy.cpp]: maybe remove the logs

**Review by rtei-pw — APPROVED** [review]
Hello all good for me. we can remove the logs if we already have a test case.

**Comment by Milos-RTEi** [comment]
I implemented the unit test. Just to save you some time if you are going to check the implementation, the procedure is:
cmake -B _build -S ./src -DCMAKE_BUILD_TYPE=debug -DBUILD_ortools=ON -DBUILD_TESTING=ON -DBUILD_OUTPUT_TEST=ON
cmake --build _build --config debug -j8
cd _build
ctest -C Release --output-on-failure

**Pull request merged**